File: day3/rucksack.py
file1 = open('input.txt')
Items = file1.readlines()

sum = 0

def getPriority(first, second):
  global sum
  commonLetters=list(set(first)&set(second))
  if (len(commonLetters)):
    letterToConvert=commonLetters[0]
    if (letterToConvert.isupper()):
      sum += ord(letterToConvert) - 38
    if (letterToConvert.islower()):
      sum += ord(letterToConvert) - 96
    


for line in Items:
  parsedLine = line.rstrip()
  lineLength = len(parsedLine)
  halfLength = lineLength//2
  firstHalf = line[0:halfLength]
  secondHalf = line[-halfLength:]
  getPriority(firstHalf, secondHalf)

print(sum)

File: day3/rucksack.pl
#!/usr/bin/perl
use warnings;
use strict;

my $file = 'input.txt';
open my $info, $file or die "could not open $file: $!";

my $result = 0;
my $groups = [];

sub get_intersection {
  print("\n");
  my $first = $_[0];
  my $second = $_[1];
  my $intersected;
  foreach my $ch (split('', $first)) {
    if (index ($second, $ch) > -1) {
      print qq/there is intersection: $ch\n/;
      $intersected = $ch;
      last;
    }
  }
  return $intersected;
}

while (my $line = <$info>) {
  chomp $line;
  my $row_length = length $line;
  my $first = substr($line, 0, $row_length/2);
  my $second = substr($line, $row_length/2, $row_length);
  my $intersected = get_intersection($first, $second);
  if ($intersected) {
    my $priority = ord($intersected);
    if ($priority > 96) {
      $result += $priority-96;
    } else {
      $result += $priority-38;
    }
  }
}

print($result);

close($file)
